cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -132,6 +132,18 @@ func (t Template) controllersPolicy() *iamv1.PolicyDocument {
 				iamv1.StringLike: map[string]string{"iam:AWSServiceName": "elasticloadbalancing.amazonaws.com"},
 			},
 		},
+		{
+			Effect: iamv1.EffectAllow,
+			Action: iamv1.Actions{
+				"iam:CreateServiceLinkedRole",
+			},
+			Resource: iamv1.Resources{
+				"arn:*:iam::*:role/aws-service-role/spot.amazonaws.com/AWSServiceRoleForEC2Spot",
+			},
+			Condition: iamv1.Conditions{
+				iamv1.StringLike: map[string]string{"iam:AWSServiceName": "spot.amazonaws.com"},
```

Comment:
ec2 was what I originally had but it was getting 403s when testing locally. I'll investigate some more.

Reply:
Re-read the docs, you're right. The direction of the IAM is delegating to the spot instance request API the ability to use ec2, so it goes something like:

CAPA(EC2:RunInstance) --> EC2(Spot:CreateRequest) --> Spot API --> Wait for fulfillment --> Spot(EC2:RunInstance).

Reply:
I wonder if we need to expand the state machine for spot instances a fair amount, in that we need to take account of unfulfilled spot requests when an awsmachine is deleted, because there won't be a corresponding machine, but AWS must fulfill the request after it's deleted.

Reply:
We use the ec2 run-instance for requesting spot instances still rather than creating spot instance requests manually. I tried a run-instances command with a really low fulfillment price and it error-ed immediately and I didn't see an `active` unfulfilled spot request in my aws console. Unsure if it'll be different if the price is fine but there isn't capacity for the spot instance. We also have the option to set the `ValidUntil` in the spot instance options so it can expire.

Reply:
OK, let's see how it goes.